notebook/services/contents/filemanager.py

```diff
@@ -491,13 +491,28 @@ def delete_file(self, path):
         if not os.path.exists(os_path):
             raise web.HTTPError(404, u'File or directory does not exist: %s' % os_path)
 
+        def _check_trash(os_path):
+            if sys.platform in {'win32', 'darwin'}:
+                return True
+
+            # It's a bit more nuanced than this, but until we can better
+            # distinguish errors from send2trash, assume that we can only trash
+            # files on the same partition as the home directory.
+            file_dev = os.stat(os_path).st_dev
+            home_dev = os.stat(os.path.expanduser('~')).st_dev
+            return file_dev == home_dev
```

Comment:
Unfortunately, this test doesn't do what you'd hope on GPFS (under Linux).  If the directories are part of different filesets, you'll still get an `EXDEV` if you try to rename a file/dir from one to the other.